### PR TITLE
Highlight projector blocks when creative mode players hold projector items on their hands

### LINKED ISSUE #4: Add an item which can show Projector location.
When people don't know the location of a projector, Finding it is a terrible thing. we should add an item which can show the projector location.

### LINKED ISSUE #5: [Feature requests] Let projector could simulate other Blocks
In some conditions, the projector which could simulate other blocks is convenient.

### PR DESCRIPTION
**Pull request by ustc-zzzz**
Fixes #4.

Without highlighting:

With highlighting:

### CODE REVIEW

File: src/main/java/org/teacon/slides/ProjectorRenderer.java [hunk 1]
```diff
@@ -1,12 +1,29 @@
 package org.teacon.slides;
 
+import com.google.common.collect.Streams;
 import com.mojang.blaze3d.matrix.MatrixStack;
 import com.mojang.blaze3d.vertex.IVertexBuilder;
-import net.minecraft.client.renderer.IRenderTypeBuffer;
-import net.minecraft.client.renderer.Matrix4f;
-import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.*;
```
Comment (3TUSK) [inline src/main/java/org/teacon/slides/ProjectorRenderer.java]: Not an urgent matter, but no star import, thanks

File: src/main/java/org/teacon/slides/ProjectorRenderer.java [hunk 2]
```diff
@@ -43,10 +60,63 @@ public void render(ProjectorTileEntity tile, float partialTicks, MatrixStack mat
...
+
+        private static Entity createDummyEntity(ClientWorld world) {
+            // dirty hack - create a long-lived glowing entity to activate entity outline shader
+            Entity entity = new AreaEffectCloudEntity(world, 0.0, 0.0, 0.0);
+            entity.setBoundingBox(IForgeTileEntity.INFINITE_EXTENT_AABB);
+            entity.ignoreFrustumCheck = true;
+            entity.setInvisible(true);
+            entity.setGlowing(true);
+            entity.canUpdate(false);
+            return entity;
```
Comment (3TUSK) [inline src/main/java/org/teacon/slides/ProjectorRenderer.java:116]: Also make it invulnerable by calling `setInvulnerable(true)`? Will `/kill @e` affect this?

**Review by SeraphJACK — APPROVED** [review]

File: src/main/java/org/teacon/slides/ProjectorRenderer.java (hunk 1 above)
Comment (SeraphJACK) [reply src/main/java/org/teacon/slides/ProjectorRenderer.java]: Why not

File: src/main/java/org/teacon/slides/ProjectorRenderer.java (hunk 2 above)
Comment (ustc-zzzz) [reply src/main/java/org/teacon/slides/ProjectorRenderer.java:116]: `/kill @e` is a server command and affects server entities only.

However, it should indeed be attentioned whether it is possible that the client entity will come to death.

**Review by FledgeXu — COMMENTED** [review]
Maybe have better way to achive this goal?﻿
https://github.com/McJty/YouTubeModding14/blob/master/src/main/java/com/mcjty/mytutorial/client/InWorldRenderer.java

**Comment by ustc-zzzz** [comment]
#7 has been merged to 1.15-forge branch.

**Pull request closed without merging**